v2/internal/pkg/release/cincinnati.go

```diff
@@ -123,29 +125,45 @@ func (o *CincinnatiSchema) GetReleaseReferenceImages(ctx context.Context) ([]v2a
 	// before making a deep copy
 	// check that the "platform.release" field is not empty
 	if len(o.Config.Mirror.Platform.Release) > 0 {
-		copyImage := v2alpha1.CopyImageSchema{
-			Source:      o.Config.Mirror.Platform.Release,
-			Destination: "",
-			Origin:      o.Config.Mirror.Platform.Release,
-		}
-		allImages = append(allImages, copyImage)
 		// OCPBUGS-50617
-		// include signature verify and download for rc (release candidate) by digest
-		// skip ec (engineering candidate) and other releases (by tag)
+		// include signature verify and download for ga releases, rc (release candidate)
+		// and ec Iengineering candidate) by tag or digest
 		imgSpec, err := image.ParseRef(o.Config.Mirror.Platform.Release)
 		if err != nil {
 			return []v2alpha1.CopyImageSchema{}, err
 		}
 		if imgSpec.IsImageByDigestOnly() {
+			copyImage := v2alpha1.CopyImageSchema{
+				Source:      o.Config.Mirror.Platform.Release,
+				Destination: "",
+				Origin:      o.Config.Mirror.Platform.Release,
+			}
+			allImages = append(allImages, copyImage)
 			imgs, err := o.Signature.GenerateReleaseSignatures(ctx, allImages)
 			if err != nil {
 				return []v2alpha1.CopyImageSchema{}, err
 			}
 			return imgs, nil
 		} else {
-			o.Log.Warn("release image is not by digest: SKIPPING signature verification")
+			srcImage := o.Opts.Global.NewSystemContext()
+			digest, err := o.Manifest.GetDigest(context.Background(), srcImage, imgSpec.ReferenceWithTransport)
+			o.Log.Debug("digest for image %s is %s", imgSpec.Reference, digest)
+			if err != nil {
+				return []v2alpha1.CopyImageSchema{}, err
+			}
+			copyImage := v2alpha1.CopyImageSchema{
+				// TODO:This will need to change when we use sha512
+				Source:      imgSpec.Name + "@sha256:" + digest,
+				Destination: "",
+				Origin:      o.Config.Mirror.Platform.Release,
+			}
+			allImages = append(allImages, copyImage)
+			imgs, err := o.Signature.GenerateReleaseSignatures(ctx, allImages)
+			if err != nil {
+				return []v2alpha1.CopyImageSchema{}, err
+			}
+			return imgs, nil
```

Comment:
Nit: this part is the same for both `if` branches so it can be extracted out:
```go
var copyImage v2alpha1.CopyImageSchema
if imgSpec.IsImageByDigestOnly() {
    copyImage = ...
} else {
    digest, err := o.Manifest.GetDigest(ctx, o.Opts.Global.NewSystemContext(), imgSpec.ReferenceWithTransport)
    if err != nil {
        return []v2alpha1.CopyImageSchema{}, err
    }
    copyImage = ...
}
allImages = append(allImages, copyImage)
return o.Signature.GenerateReleaseSignature(ctx, allImages)
```